src/ripple/json/impl/json_value.cpp

```diff
@@ -1068,6 +1068,24 @@ Value::get ( std::string const& key,
 }
 
 Value
+Value::removeElement ( std::uint32_t index )
+{
+    JSON_ASSERT (type_ == arrayValue || index < value_.map_->size ());
+
+    if ( type_ == nullValue || index >= value_.map_->size ())
```

Comment:
'index' is unsigned, it can never meet your condition.

I realize the assert redundancy but I wanted safety in a release build. The line should read:

if ( type_ != arrayValue || index >= value_.map_->size ())